app/lib/actions/candlepin/import_pool_handler.rb

```diff
@@ -68,11 +68,12 @@ def remove_pool(pool_id)
       def reindex_consumer(message)
         if message.content['newEntity']
           uuid = JSON.parse(message.content['newEntity'])['consumer']['uuid']
-          system = ::Katello::System.find_by(:uuid => uuid)
-          if system.nil?
-            @logger.debug "skip re-indexing of non-existent content host #{uuid}"
+          subscription_facet = ::Katello::Host::SubscriptionFacet.where(:uuid => uuid).first
```

Comment:
I'm not sure it would be better, it may be preferred by some?  It generates the same query.  I tend to prefer .where().first as its more explicit as to what is happening.  I've never been a fan of  .find() vs .find_by_attr() with regard to it raising an exception for one but not the other, but that is just my opinion.